src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenParameterBuilderExtensions.cs

```diff
@@ -21,7 +21,10 @@ public static class AcquireTokenParameterBuilderExtensions
             IDictionary<string, string> extraHttpHeaders)
             where T : AbstractAcquireTokenParameterBuilder<T>
         {
-            builder.CommonParameters.ExtraHttpHeaders = extraHttpHeaders;
+            foreach (var kvp in extraHttpHeaders)
+            {
+                builder.CommonParameters.ExtraHttpHeaders.Add(kvp.Key, kvp.Value);
```

Comment:
This change addresses the statement in the issue: `Note: pls improve WithExtraQueryParameters so that it can be used multiple times, in which case values are appended.`

So the headers can be added multiple times in the flow and will be maintained. 

The `ExtraHttpHeaders` here are later added to the headers sent to ests [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs#L219). Do you mean that the headers that MSAL sends to ests like `x-app-name` can get overridden accidentally?

Reply:
@neha-bhargava yep, exactly! I was curious if someone can inadvertently replace the headers that we need for eSTS telemetry.

Reply:
That is a possibility currently, I think we should add these first and then add the common headers (the one MSAL sends to ests) so that the final set have all the expected headers.

Reply:
Reverting this since we will use extraQueryParameters